content/doc/developer/plugin-development/injecting-environment-variables.adoc

```diff
@@ -0,0 +1,67 @@
+---
+layout: developerguide
+title: How to inject environment variables
+---
+
+Injecting some env variables allows to communicate steps with shell scripts and another
+steps. To achieve it, just follow this guide.
+We will add `EXTERNAL_SYSTEM_ID` environment variable in Pipeline compatible way.
+
+### Using implementation of SimpleBuildStep
+
+To be compatible both with old-fashion Freestyle jobs and brand new Pipeline, your step should
+implement `SimpleBuildStep` interface, which provides `perform(Run<?,?>, FilePath, Launcher, TaskListener)` method.
+
+First of all we should create specific implementation of `hudson.model.Action`. This will allow
+us to attach some information to the `Run` instance.
+
+Our implementation should store all required data for populating environment variable.
+
+[source,java]
+----
+@Value <1>
+public class ExternalSystemLaunchedAction implements Action {
+    String name; <2>
+    String externalId; <3>
+}
+----
+<1> We use https://projectlombok.org/[Project lombok] to simplify examples
+<2> Part of variable name
+<3> Future content of environment variable
+
+Attaching this action to the run is straightforward - just call
+
+[source,java]
+----
+run.addAction(new ExternalSystemLaunchedAction("SERVICE", externalId));
+----
+
+inside of the `SimpleBuildStep#perform` method at the right moment.
```

Comment:
Sentences should not be broken up by code samples.

Reply:
why not?